File: atlaskv-ts-sdk/src/models/PrefixResult.ts
import { Revision } from "./Revision.js";

/**
 * An individual entry matching a prefix scan.
 */
export interface PrefixEntry {
  /** The unique key identifier. */
  key: string;
  /** The value associated with the key. */
  value: string;
  /** The modification version of the key (null if metadata is excluded). */
  version: number | null;
  /** Epoch timestamp in milliseconds indicating when the key was created (null if metadata is excluded). */
  createdAt: number | null;
  /** Epoch timestamp in milliseconds indicating when the key was last updated (null if metadata is excluded). */
  updatedAt: number | null;
  /** Remaining TTL duration in milliseconds (null if metadata is excluded). */
  ttlRemaining: number | null;
  /** Associated lease ID (null if none or metadata is excluded). */
  leaseId: string | null;
  /** List of historical revisions (null if history is excluded). */
  history: Revision[] | null;
}

/**
 * Represents the paginated results of a prefix query.
 */
export interface PrefixResult {
  /** The prefix scanned in the query. */
  prefix: string;
  /** The matching entries in the current page. */
  entries: PrefixEntry[];
  /** The total count of matching keys in the store. */
  totalCount: number;
  /** The pagination offset of the current result page. */
  offset: number;
  /** The pagination page limit. */
  limit: number;
}

File: atlaskv-java-sdk/src/main/java/com/atlaskv/sdk/models/PrefixResult.java
package com.atlaskv.sdk.models;

import java.util.List;

/**
 * Immutable paginated results from a key prefix scan query.
 *
 * @param prefix     the scanned prefix
 * @param entries    the matching prefix entries in the current page
 * @param totalCount total number of matching keys in the store
 * @param offset     pagination offset
 * @param limit      pagination page limit
 */
public record PrefixResult(
        String prefix,
        List<PrefixEntry> entries,
        int totalCount,
        int offset,
        int limit
) {
    /**
     * An individual entry matching a prefix scan.
     *
     * @param key          the key
     * @param value        the value
     * @param version      the version of the key (null if metadata excluded)
     * @param createdAt    creation epoch timestamp (null if metadata excluded)
     * @param updatedAt    update epoch timestamp (null if metadata excluded)
     * @param ttlRemaining remaining TTL duration in milliseconds (null if metadata excluded)
     * @param leaseId      associated lease ID (null if none or metadata excluded)
     * @param history      list of historical revisions (null if history excluded)
     */
    public record PrefixEntry(
            String key,
            String value,
            Long version,
            Long createdAt,
            Long updatedAt,
            Long ttlRemaining,
            String leaseId,
            List<Revision> history
    ) {}
}
